src/modules/apps/link.ts

```diff
@@ -29,7 +31,71 @@ const UPDATE_SIGN = chalk.blue('U')
 const stabilityThreshold = process.platform === 'darwin' ? 100 : 200
 const AVAILABILITY_TIMEOUT = 1000
 const N_HOSTS = 3
+const builderHubInjectedDepsTimeout = 2000  // 2 seconds
+const yarn = join(__dirname, '../../../node_modules/yarn/bin/yarn.js install --force')
+
+const resolvePackageJsonPath = (builder: string) => resolvePath(process.cwd(), `${builder}/package.json`)
+
+const resolveAppId = async (appName: string, appVersion: string) => await apps.getApp(`${appName}@${appVersion}`).then(prop('id'))
+
+const injectedDependencies = async (builderName: string, builderVersion: string, workspace: string, account: string, environment: string) => {
+  const extension = (environment === 'prod') ? 'myvtex' : 'myvtexdev'  // Remove this
+  const http = axios.create({
+    baseURL: `https://${workspace}--${account}.${extension}.com`, // change this route!!!
+    timeout: builderHubInjectedDepsTimeout,
+  })
+  try {
+    const res = await http.get(`/_v/private/builder/0/getdeps/${builderName}/${builderVersion}`)
+    return res.data
+  } catch (e) {
+    return {}
+  }
+}
 
+const appTypingsURL = async (account: string, workspace: string, environment: string, appName: string, appVersion: string, builder: string): Promise<string> => {
+  const extension = (environment === 'prod') ? 'myvtex' : 'myvtexdev'  // Remove this
+  const appId = await resolveAppId(appName, appVersion)
+  const typingsPath = /\+build/.test(appId) ? 'linked/v1' : 'v1'
+  return `https://${workspace}--${account}.${extension}.com/_v/private/typings/${typingsPath}/${appId}/${builder}` // change this route!!!!!!
+}
+
+const appsWithTypingsURLs = async (builder: string, account: string, workspace: string, environment: string, appDependencies: {[key: string]: string}) => {
+  const result: {[key: string]: string} = {}
+  for (const [appName, appVersion] of Object.entries(appDependencies)) {
+    result[appName] = await appTypingsURL(account, workspace, environment, appName, appVersion, builder)
+  }
+  return result
+}
+
+const mergeDevDeps = (currentPackageJson: {[key: string]: any}, newDevDepsEntries: {[key: string]: string}) =>
+  merge(currentPackageJson.devDependencies, newDevDepsEntries)
+
+const runYarn = () => {
+  log.info('Running yarn in ./react/')
+  process.chdir('./react')
```

Comment:
Even though this will only be run in the `react` folder for now, it seems strange that this is hard-coded. Especially since this is called from `getTypings` who has a `builder` parameter that could, at some point in the future, be other than `react`.